x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/Fork.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.plan.logical;
+
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Fork is a {@code Plan} with one child, but holds several logical subplans, e.g.
+ * {@code FORK [WHERE content:"fox" ] [WHERE content:"dog"] }
+ */
+public class Fork extends UnaryPlan {
+
+    private final List<LogicalPlan> subPlans;
```

Comment:
Keeping the subplans separate from children will cause issue in the traversals of the tree (whether analysis, verification or optimization)

Reply:
A workaround for this is to have the subplans passed a children, meaning Fork will stop being a UnaryPlan but rather an Nary (regular Plan) - this is similar to InlineJoin, which is a BinaryPlan.

Reply:
We originally modelled Fork in a way similar to InlineJoin/Stats, but then went to what is in the PR when moving to support more than 2 subqueries. Lemme see what it would take to revert back to this model.